src/libsync/propagateuploadng.cpp

```diff
@@ -387,6 +389,31 @@ void PropagateUploadFileNG::slotPutFinished()
         return;
     }
 
+    // Adjust the chunk size for the time taken.
+    //
+    // Dynamic chunk sizing is enabled if the server configured a
+    // target duration for each chunk upload.
+    double targetDuration = _propagator->account()->capabilities().desiredChunkUploadDuration();
+    if (targetDuration > 0) {
+        double uploadTime = job->msSinceStart();
+
+        auto correctedSize = static_cast<quint64>(
+                _currentChunkSize / uploadTime * targetDuration);
+
+        // There can be multiple chunk uploads going on at the same time.
+        // So don't force the chunk size to the new predicted best size
+        // and instead move it there gradually.
+        _propagator->_chunkSize = qBound(
+                _propagator->minChunkSize(),
+                (_propagator->_chunkSize + correctedSize) / 2,
```

Comment:
Why take the average and not just set `correctedSize` ?   

The comment speaks about multiple chunk upload going on, but i don't see how this is related.